ingest/expression_files/mtx.py

```diff
@@ -164,16 +176,15 @@ def transform(self):
         model_id = None
 
         # All observed cells
-        data_arrays.extend(
-            GeneExpression.create_data_arrays(
-                name=f"{self.cluster_name} Cells",
-                array_type="cells",
-                values=self.cells,
-                linear_data_type="Study",
-                linear_data_id=self.study_file_id,
-                **self.data_array_kwargs,
-            )
-        )
+        for data_array in GeneExpression.create_data_arrays(
+            name=f"{self.cluster_name} Cells",
+            array_type="cells",
+            values=self.cells,
+            linear_data_type="Study",
+            linear_data_id=self.study_file_id,
+            **self.data_array_kwargs,
+        ):
+            data_arrays.append(data_array)
```

Comment:
this section should get method extracted into a "create_all_cells_data_array()" method

Reply:
How would it be different? I believe the function signature would look the same as create_data_arrays.